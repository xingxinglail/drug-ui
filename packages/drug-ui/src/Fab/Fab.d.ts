import * as React from 'react';
import { PropTypes as ComponentPropTypes } from '..';
import { ButtonBaseProps } from '../ButtonBase';
export interface FabProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, ButtonBaseProps {
    color?: ComponentPropTypes.Color;
    size?: ComponentPropTypes.Size;
    href?: string;
    disabled?: boolean;
    ref?: React.Ref<HTMLButtonElement>;
}
declare const Fab: React.FC<FabProps>;
export default Fab;
