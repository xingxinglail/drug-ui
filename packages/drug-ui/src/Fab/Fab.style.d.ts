import { Theme } from '../styles';
export declare const styles: (theme: Theme) => Record<string, any>;
