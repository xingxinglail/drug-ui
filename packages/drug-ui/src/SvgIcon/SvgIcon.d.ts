import * as React from 'react';
export declare type Color = 'inherit' | 'primary' | 'secondary' | 'disabled';
export declare type FontSize = 'inherit' | 'default' | 'small' | 'large';
export interface SvgIconProps extends React.SVGProps<SVGSVGElement> {
    color?: Color;
    fontSize?: FontSize;
    htmlColor?: string;
    titleAccess?: string;
    viewBox?: string;
    ref?: React.Ref<SVGSVGElement>;
}
declare const SvgIcon: React.FC<SvgIconProps>;
export default SvgIcon;
