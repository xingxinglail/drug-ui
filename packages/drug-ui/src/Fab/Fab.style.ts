import { Style, Styles } from 'jss';
import { Theme } from '../styles';

const root: Style = (theme: Theme) => ({
    width: 56,
    height: 56,
    padding: 0,
    fontSize: theme.typography.fontSize.large,
    minWidth: 0,
    boxShadow: '0px 3px 5px -1px rgba(0,0,0,0.2), 0px 6px 10px 0px rgba(0,0,0,0.14), 0px 1px 18px 0px rgba(0,0,0,0.12)',
    boxSizing: 'border-box',
    minHeight: 36,
    transition: 'background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
    fontWeight: 500,
    lineHeight: 1.75,
    borderRadius: '50%',
    backgroundColor: '#e0e0e0',
    '&:hover': {
        textDecoration: 'none',
        backgroundColor: '#d5d5d5'
    },
    '&$disabled': {
        color: 'rgba(0, 0, 0, 0.26)',
        boxShadow: 'none',
        backgroundColor: 'rgba(0, 0, 0, 0.12)'
    }
});

const primary: Style = (theme: Theme) => ({
    color: theme.palette.primary.contrastText,
    backgroundColor: theme.palette.primary.main,
    '&:hover': {
        backgroundColor: '#115293'
    }
});

const secondary: Style = (theme: Theme) => ({
    color: theme.palette.secondary.contrastText,
    backgroundColor: theme.palette.secondary.main,
    '&:hover': {
        backgroundColor: '#9a0036'
    }
});

const colorInherit: Style = {
    color: 'inherit'
};

const sizeSmall: Style = {
    width: 40,
    height: 40
};

const sizeMedium: Style = {
    width: 48,
    height: 48
};

const disabled: Style = {};

export const styles = (theme: Theme): Styles => ({
    root: root(theme),
    primary: primary(theme),
    secondary: secondary(theme),
    sizeSmall,
    sizeMedium,
    colorInherit,
    disabled
});
