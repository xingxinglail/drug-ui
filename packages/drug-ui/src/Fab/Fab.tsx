import * as React from 'react';
import * as PropTypes from 'prop-types';
import { PropTypes as ComponentPropTypes } from '..';
import classnames from 'classnames';
import { capitalize } from '../utils';
import { styles } from './Fab.style';
import { createUseStyles } from '../styles';
import ButtonBase, { ButtonBaseProps } from '../ButtonBase';

export interface FabProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, ButtonBaseProps {
    color?: ComponentPropTypes.Color;
    size?: ComponentPropTypes.Size;
    disabled?: boolean;
    ref?: React.Ref<HTMLButtonElement>
}

type FabClassProps = 'root' | 'primary' | 'secondary' | 'colorInherit' | 'small' | 'medium' | 'disabled';

const name = 'Fab';

const useStyles = createUseStyles<FabClassProps>(styles, name);

const Fab: React.FC<FabProps> = React.forwardRef<HTMLButtonElement, FabProps>((props, ref) => {
    const { className, color = 'default', size = 'large', disabled, component, href, children, ...rest } = props;
    const classes = useStyles();

    const classNames = classnames(
        classes.root,
        {
            [classes[color]]: color !== 'inherit' && color !== 'default',
            [classes[`size${ capitalize(size) }`]]: size !== 'large',
            [classes.colorInherit]: color === 'inherit',
            [classes.disabled]: disabled
        }
    );

    return (
        <ButtonBase
            className={ classNames }
            component={ component }
            ref={ ref }
            href={ href }
            disabled={ disabled }
            { ...rest }>
            { children }
        </ButtonBase>
    );
});

Fab.displayName = name;

Fab.defaultProps = {
    color: 'default',
    size: 'large'
};

Fab.propTypes = {
    color: PropTypes.oneOf<ComponentPropTypes.Color>(['default', 'primary', 'secondary', 'inherit']),
    size: PropTypes.oneOf<ComponentPropTypes.Size>(['small', 'medium', 'large'])
};

export default Fab;
