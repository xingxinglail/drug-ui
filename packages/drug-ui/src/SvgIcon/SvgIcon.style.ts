import { Style, Styles } from 'jss';
import { Theme } from '../styles';

const root: Style = () => ({
    fill: 'currentColor',
    width: '1em',
    height: '1em',
    fontSize: 20,
    transition: 'fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
    userSelect: 'none'
});

const colorPrimary: Style = (theme: Theme) => ({
    color: theme.palette.primary.main
});

const colorSecondary: Style = (theme: Theme) => ({
    color: theme.palette.secondary.main
});

const colorDisabled: Style = {
    color: 'rgba(0, 0, 0, 0.26)'
};

const fontSizeInherit: Style = {
    fontSize: 'inherit'
};

const fontSizeSmall: Style = {
    fontSize: 20
};

const fontSizeLarge: Style = {
    fontSize: 35
};

export const styles = (theme: Theme): Styles => ({
    root: root(theme),
    colorPrimary: colorPrimary(theme),
    colorSecondary: colorSecondary(theme),
    colorDisabled,
    fontSizeInherit,
    fontSizeSmall,
    fontSizeLarge
});
