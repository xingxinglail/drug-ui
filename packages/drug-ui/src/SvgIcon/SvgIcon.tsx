import * as React from 'react';
import * as PropTypes from 'prop-types';
import classnames from 'classnames';
import { capitalize } from '../utils';
import { styles } from './SvgIcon.style';
import { createUseStyles } from '../styles';

export type Color = 'inherit' | 'primary' | 'secondary' | 'disabled';
export type FontSize = 'inherit' | 'default' | 'small' | 'large';

export interface SvgIconProps extends React.SVGProps<SVGSVGElement> {
    color?: Color;
    fontSize?: FontSize;
    htmlColor?: string;
    titleAccess?: string;
    viewBox?: string;
    ref?: React.Ref<SVGSVGElement>
}

type SvgIconPropsClassProps =
    'root'
    | 'colorPrimary'
    | 'colorSecondary'
    | 'colorDisabled'
    | 'fontSizeInherit'
    | 'fontSizeSmall'
    | 'fontSizeLarge';

const name = 'SvgIcon';

const useStyles = createUseStyles<SvgIconPropsClassProps>(styles, name);
const SvgIcon: React.FC<SvgIconProps> = React.forwardRef<SVGSVGElement, SvgIconProps>((props, ref) => {
    const { className, color = 'inherit', fontSize = 'default', viewBox, htmlColor, titleAccess, children, ...rest } = props;
    const classes = useStyles();

    const colorClass = classes[`color${ capitalize(color) }`];
    const fontSizeClass = classes[`fontSize${ capitalize(fontSize) }`];

    const classNames = classnames(
        classes.root,
        {
            [colorClass]: colorClass && color !== 'inherit',
            [fontSizeClass]: fontSizeClass && fontSize !== 'default'
        },
        className
    );

    return (
        <svg
            className={ classNames }
            viewBox={ viewBox }
            color={ htmlColor }
            ref={ ref }
            { ...rest }>
            { children }
            { titleAccess ? <title>{ titleAccess }</title> : null }
        </svg>
    );
});

SvgIcon.displayName = name;

SvgIcon.defaultProps = {
    color: 'inherit',
    fontSize: 'default',
    htmlColor: undefined,
    titleAccess: '',
    viewBox: '0 0 24 24'
};

SvgIcon.propTypes = {
    color: PropTypes.oneOf<Color>(['inherit', 'primary', 'secondary', 'disabled']),
    fontSize: PropTypes.oneOf<FontSize>(['inherit', 'default', 'small', 'large']),
    htmlColor: PropTypes.string,
    titleAccess: PropTypes.string,
    viewBox: PropTypes.string
};

export default SvgIcon;
