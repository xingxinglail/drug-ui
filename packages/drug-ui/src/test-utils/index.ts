export { default as Wrapper } from './Wrapper';
export { getClasses } from './getClasses';

export const wait = (delay = 50) => {
    return new Promise(resolve => setTimeout(resolve, delay));
};
