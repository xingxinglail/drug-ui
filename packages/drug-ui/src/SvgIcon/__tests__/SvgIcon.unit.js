import * as renderer from 'react-test-renderer';
import * as React from 'react';
import * as enzyme from 'enzyme';
import SvgIcon from '..';
import { Wrapper } from '@drug-ui/core/test-utils';
import Fab from '@drug-ui/core/Fab';

const mount = enzyme.mount;

describe('<SvgIcon />', () => {

    it('render successful', () => {
        const json = renderer.create(<Wrapper><SvgIcon /></Wrapper>).toJSON();
        expect(json).toMatchSnapshot();
    });

    it('可以设置 className', () => {
        const wrapper = mount(<Wrapper><SvgIcon className="Fab Fab2">Hello World</SvgIcon></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('Fab')).toBeTruthy();
        expect(classList.contains('Fab2')).toBeTruthy();
    });

    it('默认状态', () => {
        const wrapper = mount(<Wrapper><SvgIcon /></Wrapper>);
        const dom = wrapper.getDOMNode();
        const classList = dom.classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.length).toBe(1);
        expect(dom.getAttribute('viewBox')).toBe('0 0 1024 1024');
        expect(dom.getAttribute('color')).toBeNull();
    });

    it('设置 color = primary', () => {
        const wrapper = mount(<Wrapper><SvgIcon color="primary" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.contains('DuiSvgIcon-colorPrimary')).toBeTruthy();
        expect(classList.length).toBe(2);
    });

    it('设置 color = secondary', () => {
        const wrapper = mount(<Wrapper><SvgIcon color="secondary" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.contains('DuiSvgIcon-colorSecondary')).toBeTruthy();
        expect(classList.length).toBe(2);
    });

    it('设置 color = disabled', () => {
        const wrapper = mount(<Wrapper><SvgIcon color="disabled" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.contains('DuiSvgIcon-colorDisabled')).toBeTruthy();
        expect(classList.length).toBe(2);
    });

    it('设置 color = xxx', () => {
        const wrapper = mount(<Wrapper><SvgIcon color="xxx" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.contains('DuiSvgIcon-colorPrimary')).toBeFalsy();
        expect(classList.contains('DuiSvgIcon-colorSecondary')).toBeFalsy();
        expect(classList.contains('DuiSvgIcon-colorDisabled')).toBeFalsy();
        expect(classList.length).toBe(1);
    });

    it('设置 fontSize = inherit', () => {
        const wrapper = mount(<Wrapper><SvgIcon fontSize="inherit" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.contains('DuiSvgIcon-fontSizeInherit')).toBeTruthy();
        expect(classList.length).toBe(2);
    });

    it('设置 fontSize = small', () => {
        const wrapper = mount(<Wrapper><SvgIcon fontSize="small" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.contains('DuiSvgIcon-fontSizeSmall')).toBeTruthy();
        expect(classList.length).toBe(2);
    });

    it('设置 fontSize = large', () => {
        const wrapper = mount(<Wrapper><SvgIcon fontSize="large" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.contains('DuiSvgIcon-fontSizeLarge')).toBeTruthy();
        expect(classList.length).toBe(2);
    });

    it('设置 fontSize = xxx', () => {
        const wrapper = mount(<Wrapper><SvgIcon fontSize="xxx" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiSvgIcon-root')).toBeTruthy();
        expect(classList.length).toBe(1);
    });

    it('设置 viewBox = 0 0 1208 1208', () => {
        const wrapper = mount(<Wrapper><SvgIcon viewBox="0 0 1208 1208" /></Wrapper>);
        const dom = wrapper.getDOMNode();
        expect(dom.getAttribute('viewBox')).toBe('0 0 1208 1208');
    });

    it('设置 htmlColor = blue', () => {
        const wrapper = mount(<Wrapper><SvgIcon htmlColor="blue" /></Wrapper>);
        const dom = wrapper.getDOMNode();
        expect(dom.getAttribute('color')).toBe('blue');
    });

    it('设置 titleAccess = svg', () => {
        const wrapper = mount(<Wrapper><SvgIcon titleAccess="svg" /></Wrapper>);
        const dom = wrapper.getDOMNode();
        const titleDom = dom.children[0];
        expect(titleDom.tagName).toBe('title');
        expect(titleDom.innerHTML).toBe('svg');
    });
});
