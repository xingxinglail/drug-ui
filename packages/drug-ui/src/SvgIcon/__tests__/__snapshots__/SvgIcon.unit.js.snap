// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgIcon /> render successful 1`] = `
<svg
  className="DuiSvgIcon-root"
  viewBox="0 0 1024 1024"
/>
`;
